### ISSUE TITLE
Linking tests with SwiftWasm 5.5 snapshots requires additional linker flags

### ISSUE THREAD

**Issue by MaxDesiatov** [swiftwasm/swift#3891]
Currently tests fail to build without additional flags passed to `swift build` when using `wasm-5.5-SNAPSHOT-2021-11-16-a` toolchain. The required flags are `-Xlinker -licuuc -Xlinker -licui18n`, there weren't previously needed with SwiftWasm 5.4.

Without them we get these linker errors:

```
wasm-ld: error: /Users/maxd/Library/Developer/Toolchains/swift-wasm-5.5-SNAPSHOT-2021-11-16-a.xctoolchain
/usr/lib/swift_static/wasi/libFoundation.a(CFRegularExpression.c.obj): undefined symbol: uregex_useAnchoringBounds_64
wasm-ld: error: /Users/maxd/Library/Developer/Toolchains/swift-wasm-5.5-SNAPSHOT-2021-11-16-a.xctoolchain
/usr/lib/swift_static/wasi/libFoundation.a(CFRegularExpression.c.obj): undefined symbol: uregex_close_64
wasm-ld: error: /Users/maxd/Library/Developer/Toolchains/swift-wasm-5.5-SNAPSHOT-2021-11-16-a.xctoolchain
/usr/lib/swift_static/wasi/libFoundation.a(CFRegularExpression.c.obj): undefined symbol: uregex_close_64
wasm-ld: error: /Users/maxd/Library/Developer/Toolchains/swift-wasm-5.5-SNAPSHOT-2021-11-16-a.xctoolchain
/usr/lib/swift_static/wasi/libFoundation.a(CFStringTransform.c.obj): undefined symbol: utrans_trans_64
wasm-ld: error: /Users/maxd/Library/Developer/Toolchains/swift-wasm-5.5-SNAPSHOT-2021-11-16-a.xctoolchain
/usr/lib/swift_static/wasi/libFoundation.a(CFStringTransform.c.obj): undefined symbol: utrans_close_64
wasm-ld: error: /Users/maxd/Library/Developer/Toolchains/swift-wasm-5.5-SNAPSHOT-2021-11-16-a.xctoolchain
/usr/lib/swift_static/wasi/libFoundation.a(CFStringTransform.c.obj): undefined symbol: utrans_openU_64
wasm-ld: error: /Users/maxd/Library/Developer/Toolchains/swift-wasm-5.5-SNAPSHOT-2021-11-16-a.xctoolchain
/usr/lib/swift_static/wasi/libFoundation.a(CFStringTransform.c.obj): undefined symbol: utrans_openU_64
wasm-ld: error: /Users/maxd/Library/Developer/Toolchains/swift-wasm-5.5-SNAPSHOT-2021-11-16-a.xctoolchain
/usr/lib/swift_static/wasi/libFoundation.a(CFStringTransform.c.obj): undefined symbol: utrans_openU_64
wasm-ld: error: /Users/maxd/Library/Developer/Toolchains/swift-wasm-5.5-SNAPSHOT-2021-11-16-a.xctoolchain
/usr/lib/swift_static/wasi/libFoundation.a(CFStringTransform.c.obj): undefined symbol: utrans_openU_64
wasm-ld: error: /Users/maxd/Library/Developer/Toolchains/swift-wasm-5.5-SNAPSHOT-2021-11-16-a.xctoolchain
/usr/lib/swift_static/wasi/libFoundation.a(CFStringTransform.c.obj): undefined symbol: utrans_close_64
wasm-ld: error: /Users/maxd/Library/Developer/Toolchains/swift-wasm-5.5-SNAPSHOT-2021-11-16-a.xctoolchain
/usr/lib/swift_static/wasi/libFoundation.a(CFStringTransform.c.obj): undefined symbol: utrans_close_64
...
```

**Comment by MaxDesiatov** [comment]
@kateinoigakukun could this be related to recent autolinking changes?

**Comment by MaxDesiatov** [comment]
Ah, this could be one of the manifestations of https://github.com/swiftwasm/swift/issues/3707, right?

**Comment by kateinoigakukun** [comment]
@MaxDesiatov 5.5 branch doesn't have `-public-autolink-library` feature. It's only available in main branch, so this issue is not fixed in 5.5 toolchain.

**Comment by MaxDesiatov** [comment]
Ok, since I can easily work around this by passing these flags in `carton`, I'm not going to dig into this just for 5.5. But will keep this open for posterity, at least until we release 5.6 or whatever the next version, just to make sure it's fixed in those later versions.
